tests/test_dataportal.py

```diff
@@ -0,0 +1,371 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from datetime import timedelta
+import bcolz
+import numpy as np
+import pandas as pd
+
+from unittest import TestCase
+from pandas.tslib import normalize_date
+from testfixtures import TempDirectory
+from zipline.data.data_portal import DataPortal
+from zipline.data.us_equity_pricing import (
+    SQLiteAdjustmentWriter,
+    SQLiteAdjustmentReader,
+    BcolzDailyBarReader
+)
+from zipline.finance.trading import TradingEnvironment, SimulationParameters
+from zipline.data.future_pricing import FutureMinuteReader
+from zipline.data.us_equity_minutes import (
+    MinuteBarWriterFromDataFrames,
+    BcolzMinuteBarReader
+)
+from .utils.daily_bar_writer import DailyBarWriterFromDataFrames
+
+
+class TestDataPortal(TestCase):
+    def test_forward_fill_minute(self):
+        tempdir = TempDirectory()
+        try:
+            env = TradingEnvironment()
+            env.write_data(
+                equities_data={
+                    0: {
+                        'start_date': pd.Timestamp("2015-09-28", tz='UTC'),
+                        'end_date': pd.Timestamp("2015-09-29", tz='UTC')
+                        + timedelta(days=1)
+                    }
+                }
+            )
+
+            minutes = env.minutes_for_days_in_range(
+                start=pd.Timestamp("2015-09-28", tz='UTC'),
+                end=pd.Timestamp("2015-09-29", tz='UTC')
+            )
+
+            df = pd.DataFrame({
+                # one missing bar, then 200 bars of real data,
+                # then 1.5 days of missing data
+                "open": np.array([0] + list(range(0, 200)) + [0] * 579)
+                * 1000,
+                "high": np.array([0] + list(range(1000, 1200)) + [0] * 579)
+                * 1000,
+                "low": np.array([0] + list(range(2000, 2200)) + [0] * 579)
+                * 1000,
+                "close": np.array([0] + list(range(3000, 3200)) + [0] * 579)
+                * 1000,
+                "volume": [0] + list(range(4000, 4200)) + [0] * 579,
+                "minute": minutes
+            })
+
+            MinuteBarWriterFromDataFrames(
+                pd.Timestamp('2002-01-02', tz='UTC')).write(
+                    tempdir.path, {0: df})
+
+            sim_params = SimulationParameters(
+                period_start=minutes[0],
+                period_end=minutes[-1],
+                data_frequency="minute",
+                env=env,
+            )
+
+            equity_minute_reader = BcolzMinuteBarReader(tempdir.path)
+
+            dp = DataPortal(
+                env,
+                equity_minute_reader=equity_minute_reader,
+                sim_params=sim_params
```

Comment:
The main reason it is passed is to get the `data_frequency` of the DataPortal, which determines the source for the data returned by `get_spot_value`.

`get_spot_value` is used on the lazy-mainline branch to pull the current price for the blotter, performance_tracker, and performance_period. That concept of current price is tied to whether the simulation is daily or minute.

Any suggestions for how to model that better?